Security/EmailProvider.php

```diff
@@ -0,0 +1,23 @@
+<?php
+
+/*
+ * This file is part of the FOSUserBundle package.
+ *
+ * (c) FriendsOfSymfony <http://friendsofsymfony.github.com/>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace FOS\UserBundle\Security;
+
+class EmailProvider extends UserProvider

```

Comment:
Existing email provider fetches user by username and by email which is corrent. However does not let people skip username itself.